### PR TITLE
 feat : homepage added

### PR DESCRIPTION
**Pull request by vikrantkode**
# Change logs

### css is too bad for card positioning

image

image

### CODE REVIEW

File: my-app/src/components/Nav/Nav.css [hunk 1]
```diff
@@ -0,0 +1,40 @@
+:root {
+  --white-color: white;
+  --header-style-color:black;
```
Comment (sharmayatendra) [inline my-app/src/components/Nav/Nav.css:3]: avoid hard coded colors

File: my-app/src/components/Nav/Nav.css [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+.header {
+  grid-area: head;
+  padding: 0px 0 0 28px;
```
Comment (sharmayatendra) [inline my-app/src/components/Nav/Nav.css:7]: use rems when value is more than 10

File: my-app/src/components/VideoCard/ViCard.css [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+
+.parent{
+    /* grid-area: main; */
+    margin-top: 4rem;
+    width: 20rem;
+    border-radius: 0.5rem; 
+    /* display: flex;   
+    justify-content: center;
+    align-items: center; */
```
Comment (sharmayatendra) [inline my-app/src/components/VideoCard/ViCard.css]: remove commented code

File: my-app/src/components/VideoCard/ViCard.css [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    flex-direction: column; 
+    box-shadow: 2px 2px 10px black;
```
Comment (sharmayatendra) [inline my-app/src/components/VideoCard/ViCard.css]: avoid hard coded color use variable for that

File: my-app/src/components/VideoCard/ViCard.css [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+.card {
+    display: flex;
+    align-items: center;
+    /* justify-content: center; */
+    /* flex-wrap: wrap; */
```
Comment (sharmayatendra) [inline my-app/src/components/VideoCard/ViCard.css]: same as above

File: my-app/src/components/VideoCard/ViCard.css [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  
+}
+
+.title__header{
+    padding-bottom: 4px;
+}
+
+.title__subheading,
+.content__color {
+    color : grey;
```
Comment (sharmayatendra) [inline my-app/src/components/VideoCard/ViCard.css]: avoid hard coded color

File: my-app/src/pages/Homepage/Homepage.css [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  
+    grid-template-columns: repeat(4, 1fr);
+    grid-template-areas:
+      /* "head     head    head    head    head    head    head    head  "
+      "section  section section section section section section section"
+      "......   main    main    main    main    main    main    ......"
+      "......   main    main    main    main    main    main    ......"
+      "......   main    main    main    main    main    main    ......"
+      "......   main    main    main    main    main    main    ......"
+      "footer   footer  footer  footer  footer  footer  footer  footer"; */
```
Comment (sharmayatendra) [inline my-app/src/pages/Homepage/Homepage.css]: same as above

**Review by sharmayatendra — APPROVED** [review]
suggested the changes can look into it

good file structure
good naming convention

File: my-app/src/components/Nav/Nav.css (hunk 2 above)
Comment (vikrantkode) [reply my-app/src/components/Nav/Nav.css:7]: Okay. will change it during refactoring.

File: my-app/src/components/VideoCard/ViCard.css (hunk 3 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: done

File: my-app/src/components/VideoCard/ViCard.css (hunk 4 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: changed it

File: my-app/src/components/VideoCard/ViCard.css (hunk 5 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: removed

File: my-app/src/pages/Homepage/Homepage.css (hunk 7 above)
Comment (vikrantkode) [reply my-app/src/pages/Homepage/Homepage.css]: removed this code

File: my-app/src/components/VideoCard/ViCard.css (hunk 6 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.css]: done

File: my-app/src/components/VideoCard/ViCard.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import "./ViCard.css"
+
+const ViCard = ({video}) => {
+    const {_id,title,category} = video;
+  return (
+    <div className="parent">
+        <div className="card">
+            <div className="image">
+                <img src={`https://i.ytimg.com/vi/${_id}/hqdefault.jpg`} alt="carosol"/>
```
Comment (sashaboi) [inline my-app/src/components/VideoCard/ViCard.jsx:10]: alt text could be something relevant like title.replaceAll(' ','-') 
this replaces all spaces and replaces with dash or you could also put underscore. SEO will be stonks.

File: my-app/src/pages/Homepage/Homepage.jsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+const Homepage = () => {
+  const [showVideoList, setShowVideoList] = useState([]);
+  useEffect(() => {  
+  (async () =>{
+   try {
+     const resp = await axios.get(`/api/videos`)
+     setShowVideoList(resp.data.videos)
+     console.log(resp.data.videos)
+   } catch (error) {
+     console.log(error)
```
Comment (sashaboi) [inline my-app/src/pages/Homepage/Homepage.jsx:19]: nishant suggested checking type of error and throwing relevant toast. 
I havent done this as well, just a suggestion for the future.

File: my-app/src/pages/Homepage/Homepage.jsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+   }
+  })();
+  },[])
+  
+  return (
+    <div className='container_home'>
+      <Nav/>
+      <div className='hero_img_container'>
+        <img className='hero_img_width' src={hero_img} alt=""  />
```
Comment (sashaboi) [inline my-app/src/pages/Homepage/Homepage.jsx:28]: alt text could be added, just nitpicking for now not very p0.

File: my-app/src/pages/Homepage/Homepage.jsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from 'react'
+import "./Homepage.css"
```
Comment (sashaboi) [inline my-app/src/pages/Homepage/Homepage.jsx:2]: A lot of people say that: even I am learning this now,

Project imports as they get longer should be divided into two. 
Local imports and library imports 

so useeffect, axios and stuff like that on top.


and components together
then context together 
then css together. 

helps debug later. 
unsure about order, read here for more : https://dev.to/otamnitram/sorting-your-imports-correctly-in-react-213m

File: my-app/src/pages/Homepage/Homepage.jsx [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+          <p className="hero_img_tagline sub_tagline">
+            Watch, Explore  and  Enjoy
+          </p>
+          <Link to="/product">
+            <button className="btn btn-default">Watch Now</button>
+          </Link>
+        </div>
+      </div>
+  
+      {showVideoList.slice(0,4).map((video)=><ViCard key={video._id} video={video}/>)}
```
Comment (sashaboi) [inline my-app/src/pages/Homepage/Homepage.jsx:40]: I dont understand what is happening here. you're trying to show just 4 videos from the total videos on homepage itself? 
but screenshots dont show this and as preview link not available , could be confusing to reviewer.

**Review by sashaboi — APPROVED** [review]
Looks amazing, 
I have done a little nitpicking , have a look at suggestions,
looking forward to seeing rest of your project.

File: my-app/src/pages/Homepage/Homepage.css [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+.container_home {
+    display: grid;
+    column-gap:20px;
+    grid-template-columns: repeat(5, 1fr);
+    grid-template-areas:
+     
+      "head    head    head     head    head"
+      "section section section section  section"
```
Comment (sashaboi) [inline my-app/src/pages/Homepage/Homepage.css:13]: image
This space left is pointless, could do better layouting , as a personal preference I am not a fan of this structure.

File: my-app/src/components/VideoCard/ViCard.jsx (hunk 8 above)
Comment (vikrantkode) [reply my-app/src/components/VideoCard/ViCard.jsx:10]: sure will look into this

File: my-app/src/pages/Homepage/Homepage.jsx (hunk 9 above)
Comment (vikrantkode) [reply my-app/src/pages/Homepage/Homepage.jsx:19]: didnt have toast for that as of now. but moving forward . will add it

File: my-app/src/pages/Homepage/Homepage.jsx (hunk 10 above)
Comment (vikrantkode) [reply my-app/src/pages/Homepage/Homepage.jsx:28]: oh..I missed.. will do that

File: my-app/src/pages/Homepage/Homepage.jsx (hunk 11 above)
Comment (vikrantkode) [reply my-app/src/pages/Homepage/Homepage.jsx:2]: okay...will rearrange them

File: my-app/src/pages/Homepage/Homepage.css (hunk 13 above)
Comment (vikrantkode) [reply my-app/src/pages/Homepage/Homepage.css:13]: i Was doing with grid now will change to flex

**Pull request merged**